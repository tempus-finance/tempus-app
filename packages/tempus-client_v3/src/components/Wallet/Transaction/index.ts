import TransactionList from './TransactionList';
import TransactionItem from './TransactionItem';

export { TransactionList, TransactionItem };
