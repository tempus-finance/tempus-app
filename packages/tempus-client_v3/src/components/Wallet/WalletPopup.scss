.tc__wallet-popup {
  position: relative;
  align-self: flex-end;

  // the CSS variable are here for React components to override
  --popupPosX: 0px;
  --popupPosY: 0px;
  // we want to hide backdrop until JS sets proper position for it.
  --popupVisibility: none;

  .tc__wallet-popup-backdrop {
    display: var(--popupVisibility);
    position: absolute;
    left: calc(-1 * var(--popupPosX));
    top: calc(-1 * var(--popupPosY));
    width: 100vw;
    height: 100vh;
    z-index: 9997;
  }

  .tc__tooltip {
    z-index: 9999;
    background-color: var(--walletPopupBackground);
    width: 328px;
    padding: 16px;
    transform: translate(-100%, calc(100% + 8px));
    box-sizing: border-box;
    border: none;
    border-radius: 5px;
    box-shadow: var(--downStrongShadow);
    display: flex;
    flex-wrap: wrap;

    .tc__wallet-popup__title {
      flex: 1;
    }

    .tc__wallet-popup__address {
      border: 1px solid var(--walletPopupInnerBorder);
      border-radius: 8px;
      padding: 16px;
      margin-top: 12px;
      flex: 1 100%;
    }

    .tc__wallet-popup__transaction-list {
      display: flex;
      flex-direction: column;
      justify-content: center;
      margin-top: 12px;
      box-sizing: border-box;

      .tc__accordion {
        width: 260px;
        margin: 4px;

        .tc__accordion__title :not(svg) {
          overflow: hidden;
          text-overflow: ellipsis;

          &:first-of-type {
            flex: 1 70%;
          }

          &:last-of-type {
            min-width: 24px;
            margin-right: 4px;
          }
        }

        .tc__wallet-popup__transaction-list__item-content {
          .tc__wallet-popup__transaction-list__item-content-timestamp,
          .tc__wallet-popup__transaction-list__item-content-detail,
          .tc__wallet-popup__transaction-list__item-content-protocol,
          .tc__wallet-popup__transaction-list__item-content-term {
            display: flex;
            align-content: center;
            margin: 4px 0;

            > * {
              margin: 0 2px;

              &:first-child {
                margin-left: 0;
              }

              &:last-child {
                margin-right: 0;
              }
            }
          }

          .tc__wallet-popup__transaction-list__item-content-timestamp {
            .tc__wallet-popup__transaction-list__item-content-timestamp__separator {
              width: 1px;
              height: 12px;
              background-color: var(--textDisabled);
              align-self: center;
            }

            .tc__formatted-date:first-child {
              .tc__formatted-date__separator {
                display: flex;
                flex-direction: column;
                justify-content: space-evenly;
                width: 2px;
                height: 7px;
                background-color: transparent;
                margin: 0 2px;

                &::before,
                &::after {
                  content: ' ';
                  display: block;
                  width: 2px;
                  height: 2px;
                  background-color: var(--formattedDateSeparatorHighContrast);
                }
              }
            }
          }

          .tc__wallet-popup__transaction-list__item-content-protocol {
            div {
              display: flex;
              align-content: center;
            }
          }

          .tc__wallet-popup__transaction-list__item-content-error {
            background-color: var(--walletPopupBackground);
            border: 1px solid var(--walletPopupInnerBorder);
            border-radius: 5px;
            padding: 8px;
            margin-top: 12px;
          }

          .tc__wallet-popup__transaction-list__item-content-link {
            margin-top: 16px;

            a {
              display: flex;
              align-content: center;

              > * {
                margin: 0 2px;

                &:first-child {
                  margin-left: 0;
                }

                &:last-child {
                  margin-right: 0;
                }
              }
            }
          }
        }
      }
    }

    &::before {
      content: initial;
    }

    &::after {
      content: initial;
    }
  }
}
