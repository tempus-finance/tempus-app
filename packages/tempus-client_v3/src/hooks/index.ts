export { useTokenBalance, useTokenBalances } from './useTokenBalance';
export { usePoolBalance } from './usePoolBalance';
export { useUserDepositedPools } from './useUserDepositedPools';
export * from './useFixedAprs';
export * from './useTvlData';
export * from './useTokenRates';
export * from './useWalletAddress';
export * from './useSelectedChain';
export * from './useLocale';
export * from './useUserPreferences';
export * from './usePoolViewOptions';
export * from './useConfig';
export * from './useDepositModalData';
export * from './useYieldAtMaturity';
export * from './useChainList';
export * from './usePoolList';
export * from './useTokenList';
export * from './useServicesLoaded';
export * from './useApproveToken';
export * from './useSigner';
export * from './useAppEvent';
export * from './useAllowances';
