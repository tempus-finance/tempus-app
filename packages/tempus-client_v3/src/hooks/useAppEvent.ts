import { state, useStateObservable } from '@react-rxjs/core';
import { createSignal } from '@react-rxjs/utils';
import { TempusPool } from 'tempus-core-services';

export type AppEventType = 'deposit' | 'withdraw';

export interface AppEvent {
  eventType: AppEventType;
  tempusPool: TempusPool;
  txnHash: string;
  timestamp: number;
}

const [appEvent$, setAppEvent] = createSignal<AppEvent>();
const stateAppEvent$ = state(appEvent$, null);
export const emitAppEvent = setAppEvent; // to remind to use for better naming only

export function useAppEvent(): [AppEvent | null, (value: AppEvent) => void] {
  const appEvent = useStateObservable(stateAppEvent$);
  return [appEvent, emitAppEvent];
}

export { appEvent$ };
