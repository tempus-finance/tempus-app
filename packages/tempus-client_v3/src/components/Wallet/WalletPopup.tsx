import { FC, useEffect, useMemo, useRef } from 'react';
import { useTranslation } from 'react-i18next';
import { shortenAccount } from 'tempus-core-services';
import { ButtonWrapper, Icon, Tooltip, Typography } from '../shared';

import './WalletPopup.scss';

export interface WalletPopupProps {
  open: boolean;
  address: string;
  onClose?: () => void;
}

const WalletPopup: FC<WalletPopupProps> = props => {
  const { open, address, onClose } = props;
  const { t } = useTranslation();

  const wrapperRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (wrapperRef.current) {
      const wrapperPosX = wrapperRef.current.getBoundingClientRect().x;
      const wrapperPosY = wrapperRef.current.getBoundingClientRect().y;
      wrapperRef.current.style.setProperty('--popupPosX', `${wrapperPosX}px`);
      wrapperRef.current.style.setProperty('--popupPosY', `${wrapperPosY}px`);
      // Show backdrop after position for it is set
      wrapperRef.current.style.setProperty('--popupVisibility', 'block');
    }
  });

  const popupContent = useMemo(
    () => (
      <>
        <Typography className="tc__wallet-popup__title" variant="body-primary" weight="bold">
          {t('WalletPopup.title')}
        </Typography>
        <ButtonWrapper onClick={onClose}>
          <Icon variant="close" size="small" />
        </ButtonWrapper>
        <div className="tc__wallet-popup__address">
          <Typography className="tc__wallet-popup__address-title" variant="body-primary" weight="medium">
            {t('WalletPopup.titleAddress')}
          </Typography>
          <Typography variant="body-primary" type="mono" weight="medium">
            {shortenAccount(address)}
          </Typography>
        </div>
      </>
    ),
    [t, onClose, address],
  );

  return (
    <div className="tc__wallet-popup" ref={wrapperRef}>
      {open && <div className="tc__wallet-popup-backdrop" onClick={onClose} />}
      <Tooltip open={open} placement="bottom-center">
        {popupContent}
      </Tooltip>
    </div>
  );
};

export default WalletPopup;
