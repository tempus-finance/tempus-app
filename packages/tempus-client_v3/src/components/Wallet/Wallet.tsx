import { FC, useCallback, useEffect, useMemo, useState } from 'react';
import ledgerModule from '@web3-onboard/ledger';
import gnosisModule from '@web3-onboard/gnosis';
import injectedModule from '@web3-onboard/injected-wallets';
import { init, useConnectWallet, useWallets } from '@web3-onboard/react';
import {
  DecimalUtils,
  ethereumChainIdHex,
  ethereumForkChainIdHex,
  fantomChainIdHex,
  ZERO_ADDRESS,
} from 'tempus-core-services';
import { WalletButton } from '../shared';
import ChainSelector from '../ChainSelector';
import { setWalletAddress, useSelectedChain, useTokenBalance } from '../../hooks';

// TODO - Check with designers if block native UI for wallet management is fine to use

import tempusLogo from './svg/Logo.svg'; // TODO - Check with designers if logo and icons are fine.
import tempusIcon from './png/Icon.png'; // TODO - Replace with svg image

// Fetch wallet modules
const injected = injectedModule();
const ledger = ledgerModule();
const gnosis = gnosisModule();

// Init wallet library before using it
init({
  wallets: [injected, ledger, gnosis], // List of wallets we want to support
  chains: [
    // List of chains we want to support
    {
      id: ethereumChainIdHex,
      token: 'ETH',
      label: 'Ethereum Mainnet',
      rpcUrl: process.env.REACT_APP_ETHEREUM_RPC || '',
    },
    {
      id: fantomChainIdHex,
      token: 'FTM',
      label: 'Fantom',
      rpcUrl: process.env.REACT_APP_FANTOM_RPC || '',
    },
    {
      id: ethereumForkChainIdHex,
      token: 'ETH',
      label: 'Ethereum Fork',
      rpcUrl: process.env.REACT_APP_ETHEREUM_FORK_RPC || '',
    },
  ],
  appMetadata: {
    name: 'Tempus App',
    logo: tempusLogo,
    icon: tempusIcon,
    description: 'Official Tempus App',
    recommendedInjectedWallets: [{ name: 'MetaMask', url: 'https://metamask.io' }],
    agreement: {
      version: '1.0',
      privacyUrl: 'https://tempus.finance/privacy-policy',
      termsUrl: 'https://tempus.finance/terms-of-service',
    },
  },
  accountCenter: {
    desktop: {
      enabled: false,
    },
  },
});

const Wallet: FC = () => {
  const connectedWallets = useWallets();
  const [{ wallet }, connect] = useConnectWallet();
  const selectedChain = useSelectedChain();

  const nativeTokenBalance = useTokenBalance(ZERO_ADDRESS, selectedChain);

  const [chainSelectorOpen, setChainSelectorOpen] = useState<boolean>(false);

  useEffect(() => {
    if (wallet) {
      setWalletAddress(wallet.accounts[0].address);
    }
  }, [wallet]);

  // TODO - Delete local storage under 'connectedWallets' when user disconnects the wallet

  /**
   * When user clicks on connect wallet button show a modal with all available wallets users can connect.
   */
  const onConnectWallet = useCallback(() => {
    connect({});
  }, [connect]);

  const onOpenChainSelector = useCallback(() => {
    setChainSelectorOpen(true);
  }, []);

  const onCloseChainSelector = useCallback(() => {
    setChainSelectorOpen(false);
  }, []);

  /**
   * Every time list of connected wallets changes, we want to store labels of those wallets in local storage.
   * Next time user opens the app, we will use this data to auto-connect wallet for the user.
   */
  useEffect(() => {
    const connectedWalletLabels = connectedWallets.map(connectedWallet => connectedWallet.label);

    if (connectedWalletLabels.length > 0) {
      window.localStorage.setItem('connectedWallets', JSON.stringify(connectedWalletLabels));
    }
  }, [connectedWallets]);

  /**
   * Check if user already connected any of his wallets in previous sessions, if yes,
   * automatically connect first wallet from list of previously connected wallets.
   */
  useEffect(() => {
    const previouslyConnectedWallets = JSON.parse(window.localStorage.getItem('connectedWallets') || '[]') as string[];
    if (previouslyConnectedWallets && previouslyConnectedWallets.length > 0) {
      connect({
        autoSelect: {
          disableModals: true,
          label: previouslyConnectedWallets[0],
        },
      });
    }
  }, [connect]);

  const walletAddress = useMemo(() => {
    if (!wallet) {
      return null;
    }
    return wallet.accounts[0].address;
  }, [wallet]);

  const balance = useMemo(() => {
    if (!nativeTokenBalance) {
      return null;
    }

    // TODO - Add number of decimals for chain native token in the config and use it here.
    return DecimalUtils.formatToCurrency(nativeTokenBalance, 2);
  }, [nativeTokenBalance]);

  return (
    <>
      <WalletButton
        address={walletAddress || ''}
        balance={balance || ''}
        chain={selectedChain ?? 'unsupported'}
        onConnect={onConnectWallet}
        onNetworkClick={onOpenChainSelector}
        // TODO - Add wallet popup
        onWalletClick={() => {}}
      />
      <ChainSelector open={chainSelectorOpen} onClose={onCloseChainSelector} />
    </>
  );
};
export default Wallet;
