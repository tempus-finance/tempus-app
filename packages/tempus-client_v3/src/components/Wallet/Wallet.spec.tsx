import { fireEvent, render, waitFor } from '@testing-library/react';
import { useConnectWallet, useSetChain } from '@web3-onboard/react';
import { BrowserRouter } from 'react-router-dom';
import { Chain, chainNameToHexChainId, Decimal, ZERO_ADDRESS } from 'tempus-core-services';
import { useSelectedChain, useTokenBalance } from '../../hooks';
import Wallet, { WalletProps } from './Wallet';

const mockConnect = jest.fn();
const mockSetChain = jest.fn();

const subject = (props: WalletProps) =>
  render(
    <BrowserRouter>
      <Wallet {...props} />
    </BrowserRouter>,
  );

jest.mock('@web3-onboard/ledger', () =>
  jest.fn().mockImplementation(() => () => ({
    label: '',
    getIcon: () => new Promise<string>(() => ''),
    getInterface: () => null,
  })),
);

jest.mock('@web3-onboard/gnosis', () =>
  jest.fn().mockImplementation(() => () => ({
    label: '',
    getIcon: () => new Promise<string>(() => ''),
    getInterface: () => null,
  })),
);

jest.mock('@web3-onboard/injected-wallets', () =>
  jest.fn().mockImplementation(() => () => ({
    label: '',
    getIcon: () => new Promise<string>(() => ''),
    getInterface: () => null,
  })),
);

jest.mock('@web3-onboard/react', () => ({
  init: jest.fn(),
  useConnectWallet: jest.fn(),
  useSetChain: jest.fn(),
  useWallets: jest.fn().mockReturnValue([]),
}));

jest.mock('react-blockies', () => () => (
  <canvas className="identicon" width="24" height="24" style={{ width: '24px', height: '24px' }} />
));

jest.mock('../../hooks', () => ({
  ...jest.requireActual('../../hooks'),
  useSelectedChain: jest.fn(),
  useTokenBalance: jest.fn(),
}));

describe('Wallet', () => {
  beforeEach(() => {
    (useConnectWallet as jest.Mock).mockReturnValue([{ wallet: null }, mockConnect]);
    (useSetChain as jest.Mock).mockReturnValue([{}, mockSetChain]);
  });

  it('renders a "Connect Wallet" button', () => {
    const { container } = subject({});

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('renders a wallet button', () => {
    (useConnectWallet as jest.Mock).mockImplementation(() => [
      { wallet: { accounts: [{ address: '0x123123123' }] } },
      () => {},
    ]);
    (useSelectedChain as jest.Mock).mockReturnValue('ethereum' as Chain);
    (useTokenBalance as jest.Mock).mockReturnValue({
      address: ZERO_ADDRESS,
      balance: new Decimal(10.456),
      chain: 'ethereum',
    });

    const { container } = subject({});

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('redirects after connect', async () => {
    const chain = 'ethereum';
    const redirectTo = `/pool/${chain}/ETH/lido`;
    const { getByRole } = subject({ redirectTo });

    const connectButton = getByRole('button');

    fireEvent.click(connectButton);

    expect(mockConnect).toBeCalledTimes(1);

    await waitFor(() => {
      expect(mockSetChain).toBeCalledTimes(1);
      expect(mockSetChain).toBeCalledWith({
        chainId: chainNameToHexChainId(chain),
      });

      expect(window.location.pathname).toBe(redirectTo);
    });
  });

  it('does not change chain if the redirect path is invalid', async () => {
    const redirectTo = '/pool/invalid/lido';
    const { getByRole } = subject({ redirectTo });

    const connectButton = getByRole('button');

    fireEvent.click(connectButton);

    await waitFor(() => {
      expect(mockConnect).toBeCalledTimes(1);
      expect(mockSetChain).not.toBeCalled();
      expect(window.location.pathname).toBe(redirectTo);
    });
  });

  it('shows chain selector on chain switcher click and hides it after closing the modal', async () => {
    (useConnectWallet as jest.Mock).mockImplementation(() => [
      { wallet: { accounts: [{ address: '0x123123123' }] } },
      () => {},
    ]);
    (useSetChain as jest.Mock).mockReturnValue([{ connectedChain: 'ethereum' }, mockSetChain]);

    const { container } = subject({});
    const chainSwitcherButton = container.querySelector('.tc__walletButton__connected-network');

    expect(chainSwitcherButton).not.toBeNull();

    if (chainSwitcherButton) {
      fireEvent.click(chainSwitcherButton);
    }

    const chainSelectorModal = container.querySelector('.tc__chainSelector');
    let closeModalButton = container.querySelector('.tc__chainSelector .tc__modal-header .tc__btn');

    expect(chainSelectorModal).not.toBeNull();
    expect(chainSelectorModal).toMatchSnapshot();

    expect(closeModalButton).not.toBeNull();

    if (closeModalButton) {
      fireEvent.click(closeModalButton);
    }

    closeModalButton = container.querySelector('.tc__chainSelector .tc__modal-header .tc__btn');

    expect(closeModalButton).toBeNull();
  });
});
