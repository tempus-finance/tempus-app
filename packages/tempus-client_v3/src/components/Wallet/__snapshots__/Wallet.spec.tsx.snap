// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders a "Connect Wallet" button 1`] = `
<div>
  <div
    class="tc__walletButton__disconnected"
  >
    <button
      class="tc__btn tc__actionButton tc__actionButton-border-secondary-large tc__actionButton-background-secondary-large tc__actionButton-hover-secondary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
      data-selected="false"
      type="button"
    >
      <div
        class="tc__actionButton-status-label"
      >
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Connect Wallet
        </div>
      </div>
    </button>
  </div>
  
  <div
    class="tc__wallet-popup"
  />
</div>
`;

exports[`Wallet renders a wallet button 1`] = `
<div>
  <div
    class="tc__walletButton__connected"
  >
    <button
      class="tc__btn tc__walletButton__connected-network"
      data-selected="false"
      type="button"
    >
      <svg
        class="tc__logo tc__logo-token-ETH"
        fill="none"
        height="24"
        viewBox="0 0 40 40"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z"
          fill="#627EEA"
        />
        <path
          clip-rule="evenodd"
          d="M11.25 20.275L20.6225 5V16.0875L11.25 20.275ZM20.6225 27.4588V34.9938L11.25 22.0201L20.6225 27.4588Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          d="M20.625 25.7163L29.9962 20.275L20.625 16.09V25.7163Z"
          fill="white"
          fill-opacity="0.2"
        />
        <path
          clip-rule="evenodd"
          d="M20.625 5V16.0875L29.9962 20.275L20.625 5ZM20.625 27.4573V34.991L30.0025 22.0173L20.625 27.4573ZM20.6225 25.7163L11.25 20.275L20.6225 16.09V25.7163Z"
          fill="white"
          fill-opacity="0.6"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <button
      class="tc__btn tc__walletButton__connected-wallet"
      data-selected="false"
      type="button"
    >
      <div
        class="tc__walletButton__connected-wallet-logo"
      >
        <canvas
          class="identicon"
          height="24"
          style="width: 24px; height: 24px;"
          width="24"
        />
      </div>
      <div
        class="tc__walletButton__connected-wallet-info"
      >
        <div
          style="font-style: normal; font-size: 10px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
        >
          0x1231...23123
        </div>
        <div
          class="tc__walletButton__connected-wallet-info-balance"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'Azeret Mono', monospace; opacity: 1;"
          >
            10.45
          </div>
           
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            ETH
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="tc__wallet-popup"
  />
</div>
`;

exports[`Wallet renders a wallet popup when click on wallet button 1`] = `
<div>
  <div
    class="tc__walletButton__connected"
  >
    <button
      class="tc__btn tc__walletButton__connected-network"
      data-selected="false"
      type="button"
    >
      <svg
        class="tc__logo tc__logo-token-ETH"
        fill="none"
        height="24"
        viewBox="0 0 40 40"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z"
          fill="#627EEA"
        />
        <path
          clip-rule="evenodd"
          d="M11.25 20.275L20.6225 5V16.0875L11.25 20.275ZM20.6225 27.4588V34.9938L11.25 22.0201L20.6225 27.4588Z"
          fill="white"
          fill-rule="evenodd"
        />
        <path
          d="M20.625 25.7163L29.9962 20.275L20.625 16.09V25.7163Z"
          fill="white"
          fill-opacity="0.2"
        />
        <path
          clip-rule="evenodd"
          d="M20.625 5V16.0875L29.9962 20.275L20.625 5ZM20.625 27.4573V34.991L30.0025 22.0173L20.625 27.4573ZM20.6225 25.7163L11.25 20.275L20.6225 16.09V25.7163Z"
          fill="white"
          fill-opacity="0.6"
          fill-rule="evenodd"
        />
      </svg>
    </button>
    <button
      class="tc__btn tc__walletButton__connected-wallet"
      data-selected="false"
      type="button"
    >
      <div
        class="tc__walletButton__connected-wallet-logo"
      >
        <canvas
          class="identicon"
          height="24"
          style="width: 24px; height: 24px;"
          width="24"
        />
      </div>
      <div
        class="tc__walletButton__connected-wallet-info"
      >
        <div
          style="font-style: normal; font-size: 10px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
        >
          0x1231...23123
        </div>
        <div
          class="tc__walletButton__connected-wallet-info-balance"
        >
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'Azeret Mono', monospace; opacity: 1;"
          >
            10.45
          </div>
           
          <div
            style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            ETH
          </div>
        </div>
      </div>
    </button>
  </div>
  <div
    class="tc__wallet-popup"
    style="--popupPosX: 0px; --popupPosY: 0px; --popupVisibility: block;"
  >
    <div
      class="tc__wallet-popup-backdrop"
    />
    <div
      class="tc__tooltip tc__tooltip__bottom-center"
    >
      <div
        class="tc__wallet-popup__title"
        style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
      >
        Wallet Overview
      </div>
      <button
        class="tc__btn "
        data-selected="false"
        type="button"
      >
        <svg
          class="tc__icon tc__icon-close"
          fill="none"
          height="16"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
            fill="#222222"
          />
        </svg>
      </button>
      <div
        class="tc__wallet-popup__address"
      >
        <div
          class="tc__wallet-popup__address-title"
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Connected Wallet
        </div>
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 500; font-family: 'Azeret Mono', monospace; opacity: 1;"
        >
          0x1231...23123
        </div>
      </div>
      <div
        class="tc__wallet-popup__transaction-list"
      >
        <div
          style="font-style: normal; font-size: 12px; line-height: 16px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Your transactions will appear here...
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Wallet shows chain selector on chain switcher click and hides it after closing the modal 1`] = `
<div
  class="tc__chainSelector"
>
  <div
    class="tc__modal-backdrop"
    id="modal-backdrop"
  />
  <div
    class="tc__modal-container "
  >
    <div
      class="tc__modal tc__modal__plain tc__modal__size-small"
    >
      <div
        class="tc__modal-header"
      >
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        >
          Select Network
        </div>
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <svg
            class="tc__icon tc__icon-close"
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
              fill="#222222"
            />
          </svg>
        </button>
      </div>
      <div
        class="tc__modal__body"
      >
        <button
          class="tc__btn tc__switcher-btn"
          data-selected="false"
          type="button"
        >
          <div
            class="tc__switcher-btn__label"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Ethereum
            </div>
          </div>
          <svg
            class="tc__logo tc__logo-token-ETH"
            fill="none"
            height="24"
            viewBox="0 0 40 40"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z"
              fill="#627EEA"
            />
            <path
              clip-rule="evenodd"
              d="M11.25 20.275L20.6225 5V16.0875L11.25 20.275ZM20.6225 27.4588V34.9938L11.25 22.0201L20.6225 27.4588Z"
              fill="white"
              fill-rule="evenodd"
            />
            <path
              d="M20.625 25.7163L29.9962 20.275L20.625 16.09V25.7163Z"
              fill="white"
              fill-opacity="0.2"
            />
            <path
              clip-rule="evenodd"
              d="M20.625 5V16.0875L29.9962 20.275L20.625 5ZM20.625 27.4573V34.991L30.0025 22.0173L20.625 27.4573ZM20.6225 25.7163L11.25 20.275L20.6225 16.09V25.7163Z"
              fill="white"
              fill-opacity="0.6"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="tc__btn tc__switcher-btn"
          data-selected="false"
          type="button"
        >
          <div
            class="tc__switcher-btn__label"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Fantom
            </div>
          </div>
          <svg
            class="tc__logo tc__logo-token-FTM"
            fill="none"
            height="24"
            viewBox="0 0 40 40"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z"
              fill="#13B5EC"
            />
            <path
              clip-rule="evenodd"
              d="M21.5 16.125L26 13.5V18.75L21.5 16.125ZM26 27.375L20 30.875L14 27.375V21.25L20 24.75L26 21.25V27.375ZM14 13.5L18.5 16.125L14 18.75V13.5ZM20.75 17.375L25.25 20L20.75 22.625V17.375ZM19.25 22.625L14.75 20L19.25 17.375V22.625ZM25.25 12.25L20 15.25L14.75 12.25L20 9.125L25.25 12.25ZM12.5 11.75V28.125L20 32.375L27.5 28.125V11.75L20 7.5L12.5 11.75Z"
              fill="white"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <button
          class="tc__btn tc__switcher-btn"
          data-selected="false"
          type="button"
        >
          <div
            class="tc__switcher-btn__label"
          >
            <div
              style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Ethereum Fork
            </div>
          </div>
          <svg
            class="tc__logo tc__logo-token-ETH"
            fill="none"
            height="24"
            viewBox="0 0 40 40"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 40C31.0457 40 40 31.0457 40 20C40 8.9543 31.0457 0 20 0C8.9543 0 0 8.9543 0 20C0 31.0457 8.9543 40 20 40Z"
              fill="#627EEA"
            />
            <path
              clip-rule="evenodd"
              d="M11.25 20.275L20.6225 5V16.0875L11.25 20.275ZM20.6225 27.4588V34.9938L11.25 22.0201L20.6225 27.4588Z"
              fill="white"
              fill-rule="evenodd"
            />
            <path
              d="M20.625 25.7163L29.9962 20.275L20.625 16.09V25.7163Z"
              fill="white"
              fill-opacity="0.2"
            />
            <path
              clip-rule="evenodd"
              d="M20.625 5V16.0875L29.9962 20.275L20.625 5ZM20.625 27.4573V34.991L30.0025 22.0173L20.625 27.4573ZM20.6225 25.7163L11.25 20.275L20.6225 16.09V25.7163Z"
              fill="white"
              fill-opacity="0.6"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
